airflow/providers/google/cloud/operators/dataprep.py

```diff
@@ -0,0 +1,58 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains a Google Cloud Storage Bucket operator.
+"""
+
+from typing import Dict
+
+from airflow.models import BaseOperator
+from airflow.providers.google.cloud.hooks.dataprep import GoogleDataprepHook
+from airflow.utils.decorators import apply_defaults
+
+
+class DataprepGetJobsForJobGroupOperator(BaseOperator):
+    """
+    Get information about the batch jobs within a Cloud Dataprep job.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:DataprepGetJobsForJobGroupOperator`
+
+
+    :param job_id The ID of the job that will be requests
+    :type job_id: int
+    """
+
+    template_fields = ("job_id",)
+
+    @apply_defaults
+    def __init__(
+        self, job_id: int, *args, **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.job_id = job_id
+
+    def execute(self, context: Dict):
+        self.log.info("Fetching data for job with id: %s ...", self.job_id)
+        hook = GoogleDataprepHook()
+        response = hook.get_jobs_for_job_group(job_id=self.job_id)
+
+        self.log.info("Push response to XCOM...")
+        self.xcom_push(context, key="job_data", value=response)
+        return response
```

Comment:
Is there any reason keep whole response twice in Xcom?
User can do either
```
{{ task_instance.xcom_pull("task_id") }}
```
or
```
{{ task_instance.xcom_pull("task_id", key="job_data") }}
```
The second option is longer

Reply:
Hm, so return `response` or `self.xcom_push(context, key="job_data", value=response)`?

Reply:
```
return response == self.xcom_push(context, key="return_value", value=response)
```

Reply:
So what is better solution in your opinion? Leave `return response`, zself.xcom_push(context, key="job_data", value=response)`, or use one of the jinja templates?

Reply:
`return response` is enough unless you know that it includes a information that surfaced under another key will help users. Consider this example:

An API after creating a resource return a dict:
```
{
    "project_id": PROJECT_ID,
    "cluster_name": CLUSTER_NAME,
    "config": {
        "master_config": {
            "num_instances": 1,
            "machine_type_uri": "n1-standard-4",
            "disk_config": {"boot_disk_type": "pd-standard", "boot_disk_size_gb": 1024},
            "id" : 90129021,
        },
        "worker_config": {
            "num_instances": 2,
            "machine_type_uri": "n1-standard-4",
            "disk_config": {"boot_disk_type": "pd-standard", "boot_disk_size_gb": 1024},
        },
    },
}
``` 
And the operator return this dict. But in downstream operator in a DAG users **usually** need to use the `id` of master config. So they do:
```
master_id = "{{ task_instance.xcom_pull('task_id')['config']['master_config']['id'] }}"
```
If you know this you can add to your operator the following code:
```
self.xcom_push(context, key="master_id", value=result['config']['master_config']['id'] )
```
so they can simply do:
```
master_id = "{{ task_instance.xcom_pull('task_id', key='master_id') }}"
```

Reply:
Thank you, I needed this explicit explanation. It explains a lot.

Reply:
I will stick to the `return response`. Now I understand how `_execute_task()` works.